src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/RemoteAppServerOptions.cs

```diff
@@ -20,6 +45,6 @@ public class RemoteAppServerOptions
     /// <summary>
     /// Gets or sets an API key used to secure the endpoint
     /// </summary>
-    [Required]
+    [Required, RegularExpression(NonEmptyGuidRegex, ErrorMessage = "API Key must be 32 hex characters (for example a GUID)")]
```

Comment:
Can we check to see if it would be sufficient to just do `MinLength(16)`? That way we wouldn't be as opinionated and would probably be ensure sufficiently complex strings

Reply:
I can check. I'm inclined to merge this as is now and follow-up in parallel to see if it's ok to relax the restriction to just be 16 characters long so that this PR isn't blocked.

Reply:
Sounds good. This requirement will be a subset of a `MinLength(16)`, so would be fine to change to that later.